src/main/java/graphql/schema/GraphQLFieldDefinition.java

```diff
@@ -185,11 +163,7 @@ public Builder deprecate(String deprecationReason) {
 
         public GraphQLFieldDefinition build() {
             if (dataFetcher == null) {
-                if (isField) {
-                    dataFetcher = new FieldDataFetcher(name);
-                } else {
-                    dataFetcher = new PropertyDataFetcher(name);
-                }
+                dataFetcher = new PropertyDataFetcher(name);
```

Comment:
> Actually I don't know how one can assume that the fields or the getters will be publicly accessible

@IamCornholio my assumption is that the developer knows what kind of object is returned by `#getSource()` at every level! The flexibility provided by the current `PropertyDataFetcher` is not because the user doesn't know what type of object they're dealing with, but because `PropertyDataFetcher` is meant to work for maps, fields, and properties.

So, if you know that a field's source will be an object with a publicly accessible field, use `FieldDataFetcher`. If you know it will be a `Map`, use `MapDataFetcher`.

> I think PropertyDataFetcher is the one to go with.

@markphilpot you don't think users are returning `Map`s more often than objects with getters? `Map` is the first test in the current implementation.